src/intervalSequence.ts

```diff
@@ -1,131 +1,103 @@
 import * as sequence from './sequence'
 import * as equal from './equal'
 import * as sign from './sign'
-import * as predicate from './predicate'
 
 // [edge, +Infinity)
 export type IntervalLeft<E, T> = {
     readonly edge: E
     readonly value: T
 }
 
+export type IntervalLeftSequence<E, T> = sequence.Sequence<IntervalLeft<E, T>>
+
+export type First<T> = {
+    readonly value: T
+}
+
 export type IntervalSequence<E, T> = {
-    readonly first: T
-    readonly rest: sequence.Sequence<IntervalLeft<E, T>>
+    readonly value: First<T>
+    readonly next: () => IntervalLeftSequence<E, T>
```

Comment:
What's the reasoning behind the change from `first`/`rest` (which looks natural for a sequence if I have my Lisp glasses on) and `value`/`next`. Is there any implication in this selection of names?

Reply:
We need compatibility with `Sequence`. I'm ok with `first` and `rest` but we should also use it `Sequence`.